include/alp/storage.hpp

```diff
@@ -331,165 +447,199 @@ namespace alp {
+					> fused_row_col;
 
-				template< typename AMFType >
-				struct Transform< view::Views::transpose, AMFType > {
+					typedef typename fused_row::resulting_imf_type final_imf_r_type;
+					typedef typename fused_row_col::resulting_imf_type final_imf_c_type;
+					typedef typename fused_row_col::resulting_polynomial_type final_polynomial_type;
+
+				public:
+
+					typedef AMF< final_imf_r_type, final_imf_c_type, final_polynomial_type > amf_type;
 
 					static
-					AMF<
-						typename AMFType::imf_c_type,
-						typename AMFType::imf_r_type,
-						typename polynomials::apply_view<
-							view::Views::transpose,
-							typename AMFType::mapping_polynomial_type
-						>::type
-					>
-					Create( const AMFType &amf ) {
-						typedef typename polynomials::apply_view< view::Views::transpose, typename AMFType::mapping_polynomial_type >::type new_mapping_polynomial_type;
-						return AMF<
-							typename AMFType::imf_c_type,
-							typename AMFType::imf_r_type,
-							new_mapping_polynomial_type
-						>(
-							amf.imf_c,
-							amf.imf_r,
-							new_mapping_polynomial_type(
-								amf.map_poly.ay2, amf.map_poly.ax2, amf.map_poly.axy,
-								amf.map_poly.ay, amf.map_poly.ax,
-								amf.map_poly.a0
+					amf_type
+					Create( ViewImfR imf_r, ViewImfC imf_c, AMF< SourceImfR, SourceImfC, SourcePoly > amf ) {
+						composed_imf_r_type composed_imf_r { imf::ComposedFactory::create( imf_r, amf.imf_r ) };
+						composed_imf_c_type composed_imf_c { imf::ComposedFactory::create( imf_c, amf.imf_c ) };
+						return amf_type(
+							fused_row::CreateImf( composed_imf_r ),
+							fused_row_col::CreateImf( composed_imf_c ),
+							fused_row_col::CreatePolynomial(
+								composed_imf_c,
+								fused_row::CreatePolynomial( composed_imf_r, amf.map_poly )
 							),
 							amf.storage_dimensions
 						);
 					}
-				};
 
-				template< typename AMFType >
-				struct Transform< view::Views::diagonal, AMFType > {
+			}; // class View
 
-					static
-					AMFType
-					Create( const AMFType &amf ) {
-						return amf;
-					}
-				};
+			/**
+			 * @brief Transforms the provided AMF by applying the provided View type.
+			 *
+			 * Exposes the type of the resulting AMF and implements a factory
+			 * method that creates objects of such type.
+			 *
+			 * @tparam view       The enum value of the desired view type.
+			 * @tparam SourceAMF  The type of the target AMF
+			 *
+			 */
+			template< enum view::Views view, typename SourceAMF >
+			struct Transform {
+
+				typedef SourceAMF amf_type;
+
+				static
+				amf_type
+				Create( const SourceAMF &amf ) {
+					throw std::invalid_argument( "Not implemented for the provided view type." );
+					return amf;
+				}
+
+			}; // class Transform
+
+			template< typename SourceAMF >
+			struct Transform< view::Views::original, SourceAMF > {
+
+				typedef SourceAMF amf_type;
+
+				static
+				amf_type
+				Create( const SourceAMF &amf ) {
+					return amf;
```

Comment:
I think this is creating a copy. Is this guaranteed if so OK otherwise perhaps make an explicit copy?